protocols/357404/357404.ot2.apiv2.py

```diff
@@ -0,0 +1,518 @@
+        pip.move_to(start_location)
+        for i in range(steps):
+            loc = start_location.move(Point(0, i*dy, 0))
+            if is_verbose:
+                ctx.comment("Dispensing at: {}".format(loc))
+            pip.dispense(dv, loc)
+
+    # reagents
+    '''
+    Define where all reagents are on the deck using the labware defined above.
+
+    e.g.
+
+    water = reservoir12.wells()[-1]
+    waste = reservoir.wells()[0]
+    samples = plate.rows()[0][0]
+    dnase = tuberack.wells_by_name()['A4']
+
+    '''
+    # Each slide block has 8 "wells", each well uses up 100 uL of each
+    # reagent: Block, antibody1, antibody2, nuclear counterstain
+    block = VolTracker(tuberack, vol_reagent, start=1, end=4,
+                       msg="Refill Block reagent tubes")
+    antibody1 = VolTracker(tuberack, vol_reagent, start=5, end=8,
+                           msg="Refill Antibody 1 reagent tubes")
+    antibody2 = VolTracker(tuberack, vol_reagent, start=9, end=13,
+                           msg="Refill Antibody 2 reagent tubes")
+    # Nuclear counterstain
+    nuc_cstn = VolTracker(tuberack, vol_reagent, start=14, end=17,
+                          msg="Refill Nuclear counterstain reagent tubes")
+    pbs = VolTracker(
+        reservoir, 290*10**3, start=1, end=1, mode='reagent',
+        pip_type='single', msg="Refill PBS reservoir")
+
+    # plate, tube rack maps
+    # Add all wells up to the block before the last block
+    # Assume that they all have 8 samples
+    target_wells = []
+    for slide_block in slide_blocks[:-1]:
+        for well in slide_block.wells():
+            target_wells.append(well)
+
+    # Add the wells of the last block, samples may be less than 8
+    for i in range(n_last_samples):
+        target_wells.append(slide_blocks[-1].wells()[i])
+
+    '''
+    Define any plate or tube maps here.
+
+    e.g.
+
+    plate_wells_by_row = [well for row in plate.rows() for well in row]
+
+    '''
+
+    # protocol
+
+    '''
+
+    Include header sections as follows for each "section" of your protocol.
+
+    Section can be defined as a step in a bench protocol.
+
+    e.g.
+
+    ctx.comment('\n\nMOVING MASTERMIX TO SAMPLES IN COLUMNS 1-6\n')
+
+    for .... in ...:
+        ...
+        ...
+
+    ctx.comment('\n\nRUNNING THERMOCYCLER PROFILE\n')
+
+    ...
+    ...
+    ...
+
+
+    '''
+    # Set the temperature to 4 degrees on the temperature module.
+    temp_mod.set_temperature(4)
+    # Transfer 100 µL of block from the tuberack to each destination well
+    # (slide) (could be done as a multi-dispense with the P1000)
+    # Measure time from the start of pipetting and subtract from 1 hr pause
+    # so that the next pipetting step starts after 1 hour and not 1 hour +
+    # the time it takes to finish the reagent transfer
+    if not is_start_after_1st_incbn:  # Skip 1st incubation?
+        ctx.comment("\n\nAdding block reagent\n")
+        t = time.time()
```

Comment:
@StevenPlonk developed a good solution to this problem (set incubation times dependent on pipetting steps), but something still in progress...